iothub_client/inc/iothub_device_client_ll.h

```diff
@@ -378,6 +378,88 @@ typedef struct IOTHUB_CLIENT_CORE_LL_HANDLE_DATA_TAG* IOTHUB_DEVICE_CLIENT_LL_HA
      */
      MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_LL_UploadMultipleBlocksToBlob, IOTHUB_DEVICE_CLIENT_LL_HANDLE, iotHubClientHandle, const char*, destinationFileName, IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_CALLBACK_EX, getDataCallbackEx, void*, context);
 
+     /**
+     * @brief    This API creates a context for a new blob upload to Azure Storage.
+     * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+     *           This function is expected to be used along with:
+     *           `IoTHubDeviceClient_LL_UploadBlockToBlob`
+     *           `IoTHubDeviceClient_LL_CompleteUploadToBlob`
+     *           `IoTHubDeviceClient_LL_DestroyUploadContext`
+     *           For the standard less-granular uploads to blob please use either
+     *           `IoTHubDeviceClient_LL_UploadToBlob` or `IoTHubDeviceClient_LL_UploadMultipleBlocksToBlob`.
+     *
+     * @param    iotHubClientHandle      The handle created by a call to the create function.
+     * @param    destinationFileName     name of the file.
+     *
+     * @warning  This is a synchronous/blocking function.
+     *
+     * @return   A `IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE` on success or NULL if the function fails.
+     */
+     MOCKABLE_FUNCTION(, IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE, IoTHubDeviceClient_LL_CreateUploadContext, IOTHUB_DEVICE_CLIENT_LL_HANDLE, iotHubClientHandle, const char*, destinationFileName);
+
+     /**
+     * @brief    This API upload a single blob block to Azure Storage (performs a PUT BLOCK operation).
+     * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
```

Comment:
As above, please document what happens for error cases.

Reply:
Added link to blob service rest api.